### PR TITLE
[HeaderStackView] Add basic Snapshot tests.

### LINKED ISSUE #6815: [HeaderStackView] Add basic snapshot tests
This was filed as an internal issue. If you are a Googler, please visit [b/128033535](http://b/128033535) for more details.

---
#### Internal data
- Associated internal bug: [b/128033535](http://b/128033535)

### PR DESCRIPTION
**Pull request by romoore**
Adding simple Snapshot tests for bars with `sizeThatFits` values.

Closes #6815

### CODE REVIEW

File: components/HeaderStackView/tests/snapshot/MDCHeaderStackViewSnapshotTests.m [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  self.bottomView = [[MDCHeaderStackViewSnaphotTestDummyView alloc] init];
+  self.bottomView.backgroundColor = UIColor.magentaColor;
+
+  self.headerStackView = [[MDCHeaderStackView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
+  self.headerStackView.topBar = self.topView;
+  self.headerStackView.bottomBar = self.bottomView;
+}
+
+- (void)tearDown {
+  self.headerStackView = nil;
```
Comment (codeman7) [inline components/HeaderStackView/tests/snapshot/MDCHeaderStackViewSnapshotTests.m:81]: _Tip:_ Set `topView` and `bottomView` to `nil`

**Review by codeman7 — APPROVED** [review]
Why are we testing for _1/3_ instead of _1_?

File: components/HeaderStackView/tests/snapshot/MDCHeaderStackViewSnapshotTests.m (hunk 1 above)
Comment (romoore) [reply components/HeaderStackView/tests/snapshot/MDCHeaderStackViewSnapshotTests.m:81]: Done.

**Comment by romoore** [comment]
@codeman7 I tested 1/3 to see how it adjusts in `sizeToFit:`. I was surprised to see how it reacted with both `topBar` and `bottomBar`, but that's apparently the documented behavior.

**Pull request merged**